Makefile

```diff
@@ -0,0 +1,31 @@
+
+PREFIX ?= /usr/local
+
+WARN_FLAGS = -Wall -Wextra -Wpedantic
+OPT_FLAGS = -O3 -flto
+ALLFLAGS = -pthread # passed to both compiling/linking stages
+CFLAGS += -MD -MP -std=c11 $(WARN_FLAGS) $(OPT_FLAGS) $(ALLFLAGS)
```

Comment:
Some notes:
- `-MD -MP` allows for minimal rebuilds by generating `%.d` files included (if they exist) below.

- The reason for splitting the flags out int different variables is so you can override them on the commandline, eg. `make WARN_FLAGS=` will disable warnings.